src/functions.js

```diff
@@ -12,45 +18,117 @@ function getNumberOfBikes( document ) {
 };
 
 function getAllButtonText( document ) {
-  // get the text inside all tags with class 'btn' and return as an array of strings
-  
+  var buttonText = document.getElementsByClassName('btn');// get the text inside all tags with class 'btn'
+    var buttonList = [];
+
+    for (i=0; i<buttonText.length; i++){
```

Comment:
If you want to refactor this, you can convert this pattern to a `.map()` call on the initial array.

Every time you find yourself doing these things in succession is a good spot to use the /map pattern:
* declare a new empty array
* loop through each element of the old array and apply a transformation or function to it
* pushing the result in the new empty array